### PR TITLE
GCI-142 Check bad code practices through Findbugs or Sonar and fix those issues

### PR DESCRIPTION
**Pull request by AkshatJain1**
## Description of what I changed

Changed made according to Sonar efficiency recommendations:

- Removed superfluous exception handlers
- Simply inherited overridden methods in BaseChangeableOpenmrsData and  BaseChangeableOpenmrsMetadata classes
- Moved various comments to its own line

## Issue I worked on

see https://issues.openmrs.org/browse/GCI-142

also see the task instance:

https://codein.withgoogle.com/dashboard/task-instances/5385621989228544/

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/Allergies.java [hunk 1]
```diff
@@ -46,36 +46,36 @@ public boolean add(Allergy allergy) {
...
 	}
-	
+
 	@Override
 	public void clear() {
 		allergyStatus = UNKNOWN;
 		allergies.clear();
 	}
-	
-	public void confirmNoKnownAllergies() throws APIException {
```
Comment (samuelmale) [inline api/src/main/java/org/openmrs/Allergies.java]: Why did you remove the throws clause?

File: api/src/main/java/org/openmrs/Allergy.java [hunk 2]
```diff
@@ -305,27 +305,27 @@ public AllergyReaction getAllergyReaction(Integer allergyReactionId) {
 				return reaction;
 			}
 		}
-		
+
```
Comment (samuelmale) [inline api/src/main/java/org/openmrs/Allergy.java]: Most of your changes are just whitespaces, could you resolve that, here and in many other places?

**Comment by samuelmale** [comment]
@AkshatJain1 can you put the link to yr task instance.

cc: https://talk.openmrs.org/t/more-tips-for-making-good-pull-requests-for-gci-students/20318

File: api/src/main/java/org/openmrs/Allergies.java (hunk 1 above)
Comment (AkshatJain1) [reply api/src/main/java/org/openmrs/Allergies.java]: I did this because an APIException is a RuntimeException which is already automatically thrown, hence making the throws specification superfluous. 

Reference: https://sonar.openmrs.org/project/issues?id=org.openmrs%3Aopenmrs&resolved=false

**Comment by AkshatJain1** [comment]
The description has been updated with the task instance.

File: api/src/main/java/org/openmrs/Allergies.java (hunk 1 above)
Comment (samuelmale) [reply api/src/main/java/org/openmrs/Allergies.java]: Cool, true that.

**Comment by samuelmale** [comment]
Thanks for working on this @AkshatJain1 

According your PR, most of your changes are whitespace, was that also recommended by Sonar?

**Comment by AkshatJain1** [comment]
They weren’t reccomended by Sonar. I am currently in the processing of mitigating this issue in my commit.

**Comment by dkayiwa** [comment]
@AkshatJain1 are you still working on this?
